winpr/libwinpr/utils/stream.c

```diff
@@ -96,13 +98,23 @@ wStream* Stream_New(BYTE* buffer, size_t size)
 
 	s->pool = NULL;
 	s->count = 0;
-
+	s->isStatic = FALSE;
 	return s;
 }
 
+void Stream_WrapBuffer(wStream *s, BYTE *buffer, size_t size)
+{
+	assert(buffer);
+	s->buffer = s->pointer = buffer;
```

Comment:
Might be a good idea to free previously allocated buffers if the `wStream*` was created by `Stream_New`

Reply:
My intent was that you either use `Stream_WrapBuffer` or `Stream_New`, not recycling a `wStream` retrieved from `Stream_New`. There's `Stream_SetBuffer` if you want to change where a dynamically allocated stream points.

Reply:
perhaps that the function name should be more explicit on the fact that we target a static stream, like `Stream_StaticInit` like it was before.